### PR TITLE
[split #6908 1/3] Fix realloc error handling

### PR DESCRIPTION
**Pull request by goekce**
I split my PR #6908 into three which was already partly reviewed. This is part 1/3.

### CODE REVIEW

File: src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx
Comment (stsoe) [inline src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx:219]:
```suggestion
      if (!result)
```
Checking against nullptr is just silly and C++ or C style at all.

File: src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:117]: ditto

File: src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/shim.cxx
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/shim.cxx:159]: ditto

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx
Comment (goekce) [reply src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx:219]: AFAIK `nullptr` was introduced in C++11 to distinguish it from arithmetic zero (as in `NULL`). This avoids problems when we have two overloaded functions where one function has an `int` as an argument and the other a pointer as an argument. I thought it is better to be more clear in the if statement. But [I read that `nullptr` evaluates to `false` in a boolean context](https://stackoverflow.com/a/57527724), so your solution is shorter. Thanks!

Can you elaborate on why is this silly? I am just curious.

Comment (stsoe) [reply src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx:219]: Sure it's not really a big deal.   Absolutely nullptr in place of NULL of course.

But checking against nullptr is redundant and doesn't make the code more readable, IMO.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if

**Comment by goekce** [comment]
Note that there are further expressive checks against `nullptr` in these files which seem to me as obvious null pointer checks.

**Comment by stsoe** [comment]
> Note that there are further expressive checks against `nullptr` in these files which seem to me as obvious null pointer checks.

Yes, the code is not perfect for sure :-(  I just happened to see this change.  Thanks!

File: src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx
Comment (goekce) [reply src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx]: After reading the guideline your comment makes more sense, thanks @stsoe

**Comment by goekce** [comment]
I am not very-versed with the PR reviews in Github. I have seen that Github still requires a review from @stsoe even I implemented the feedback so I rerequested a review.

**Review by stsoe — APPROVED** [review]
Thank you.  I added back @chvamshi-xilinx and @venkatp-xilinx as code owners.
Please sign-off your changes, per[ ](https://github.com/Xilinx/XRT/blob/master/CONTRIBUTING.rst)
Thank you for your efforts, much appreciated.

**Review by chvamshi-xilinx — APPROVED** [review]

**Review by venkatp-xilinx — APPROVED** [review]

**Comment by stsoe** [comment]
ok to test

**Pull request merged**
